### PR TITLE
Add 8051 disassemble/reassemble tests on realtek firmware.

### PR DESCRIPTION
**Pull request by gogo2464**
- [X] Mark this if you consider it ready to merge
- [X] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: test/Makefile
Comment (trufae) [inline test/Makefile:65]: delete those empty lines

File: test/Makefile
Comment (trufae) [inline test/Makefile:23]: use `cmp` instead. `diff` is not portable

File: test/Makefile
Comment (trufae) [inline test/Makefile:21]: make clean should remove this .asm file

File: test/Makefile
Comment (trufae) [inline test/Makefile:20]: run this target from another make target or in the ci, so we can see if the test works before mergin

**Review by trufae — CHANGES_REQUESTED** [review]
.

File: test/Makefile
Comment (gogo2464) [reply test/Makefile:20]: I am 100% sure it works. When I test locally it works. I also pushed a very similar test on another repo. See: https://github.com/gogo2464/realtek-firmware-decompilation/blob/master/.github/workflows/.travis.yml#L24

Comment (gogo2464) [reply test/Makefile:20]: Ah yes I just mentionned the issue with another architecture.

**Comment by gogo2464** [comment]
The http file takes too long I am not responsible of that.

File: Makefile
Comment (trufae) [inline Makefile:394]: That line those nothing. The very next line wont maintain the same environment

also its wrong to override user settings. If you have warnjngs fix your system

File: test/Makefile
Comment (trufae) [reply test/Makefile:20]: Well. You cant be sure if that will break tomorrow. Theres no sense in adding tests that never run

**Review by trufae — CHANGES_REQUESTED** [review]
Which http are you talking about

File: Makefile
Comment (gogo2464) [reply Makefile:394]: It is to ensure that local tests of everyone running tests have same setting.

**Comment by gogo2464** [comment]
By http I mean the failling test.

File: Makefile
Comment (trufae) [reply Makefile:394]: no. this line is useless. try this if my epxlanation above wasnt clear

```
$ make
export JEJE=123
echo $JEJE

$ cat Makefile
all:
	export JEJE=123
	echo $$JEJE
$
```

File: test/Makefile
Comment (trufae) [inline test/Makefile:23]: thats not valid makefile. please read the other comments and address them

**Comment by trufae** [comment]
The simplest way to move on here is just to use a shellscript in `test/test-8051-reassembly.sh` (and call it from the makefile). because makefiles are not meant to be used as shellscript replacements. the logic is different

File: Makefile
Comment (gogo2464) [reply Makefile:394]: ah ok

File: test/Makefile
Comment (gogo2464) [reply test/Makefile]: Nice catch. I decided to put the .asm file in the ./tmp folder to delete it.

**Comment by gogo2464** [comment]
Hard to test with no master running.

**Comment by trufae** [comment]
the ci is green again you can rebase if you like the green color

**Comment by gogo2464** [comment]
I prefer the red color. But I also prefer when the code is running. Then let's rebase. :)

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:1]:
```
#!/bin/sh

T=$(mktemp -d tmp.XXXXXX)
[ -z "$T" ] && exit 1
mkdir -p $T
....

rm -rf $T
```

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:5]: just compare the return code. the output may differ between implementations

```
if [ $? = 0 ]; then ,...

```

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:3]: check return code with `|| exit 1`

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:4]: check error code with || exit 1

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:2]:
```suggestion
printf "Reassembling 8051... "

```

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:11]:
```suggestion
	exit 1
```

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:7]:
```suggestion
	echo "OK"
	exit 0
```

**Review by trufae — CHANGES_REQUESTED** [review]

**Comment by gogo2464** [comment]
Ready to merge but it would be better if I did not disassemble/reassemble the offset.

File: test/Makefile
Comment (trufae) [inline test/Makefile:2]:
```suggestion
```

File: test/Makefile
Comment (trufae) [inline test/Makefile:5]: why dont run those 3 targets in parallel as it was doing before? please change this

File: test/Makefile
Comment (trufae) [inline test/Makefile:25]:
```suggestion
	$(SHELL) ./test-8051-reassembly.sh
```

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:3]: mktemp already creates the directory, this line is not necessary
```suggestion
```

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:10]:
```suggestion
if [ $? -eq 0 ]; then
```

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:11]:
```suggestion
```

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:20]: This line is never executed because you exit before leaving. use RC=0 and then exit $RC

**Review by trufae — CHANGES_REQUESTED** [review]

**Comment by gogo2464** [comment]
A bit different than expected. You could merge anyway.

**Comment by trufae** [comment]
Stop saying me to merge because your script is not even able to execute

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:3]:
```suggestion
export EXIT_CODE=0
```

**Comment by trufae** [comment]
I think it should be fine now. Lets wait for the ci. But if the script fails to run because the reassembling is not correct then i dont see the point on having this test

File: test/test-8051-reassembly.sh
Comment (trufae) [inline test/test-8051-reassembly.sh:10]:
```suggestion
if [ $? -eq 0 ]; then
```

**Review by trufae — COMMENTED** [review]
will be good if you try to run the script before pushing a change. or at least apply my suggestions instead of adding more breaking changes

**Pull request merged**
